File: app/src/main/java/com/chrisprime/primestationonecontrol/activities/BaseEventBusAppCompatActivity.java
package com.chrisprime.primestationonecontrol.activities;

import android.support.v7.app.AppCompatActivity;

import com.chrisprime.primestationonecontrol.PrimeStationOneControlApplication;
import com.squareup.otto.Bus;

/**
 * Created by cpaian on 7/26/15.
 */
public class BaseEventBusAppCompatActivity extends AppCompatActivity {
    protected Bus mEventBus = PrimeStationOneControlApplication.getEventBus();

    @Override
    public void onResume() {
        super.onResume();
        mEventBus.register(this);
    }

    @Override
    public void onPause() {
        super.onPause();
        mEventBus.unregister(this);
    }
}

File: app/src/main/java/com/chrisprime/primestationonecontrol/activities/BaseEventBusAppCompatActivity.kt
package com.chrisprime.primestationonecontrol.activities

import android.support.v7.app.AppCompatActivity

import com.chrisprime.primestationonecontrol.PrimeStationOneControlApplication
import com.chrisprime.primestationonecontrol.model.PrimeStationOne
import com.squareup.otto.Bus

/**
 * Created by cpaian on 7/26/15.
 */
open class BaseEventBusAppCompatActivity : AppCompatActivity() {
    protected var mEventBus = PrimeStationOneControlApplication.eventBus

    public override fun onResume() {
        super.onResume()
        mEventBus.register(this)
    }

    public override fun onPause() {
        super.onPause()
        mEventBus.unregister(this)
    }

    fun getCurrentPrimeStationOne(): PrimeStationOne? {
        return (application as PrimeStationOneControlApplication).currentPrimeStationOne
    }
}

